# Load data, clean 100% missing columns

In [14]:
import pandas as pd

df = pd.read_csv('listings.csv', low_memory=False)

print(df.shape)

# check columns with 100% missing values
total_rows = df.shape[0]
total_missing = df.isnull().sum()

total_missing[total_missing == total_rows].index

# drop the columns with 100% missing values
new_df = df.drop(columns=total_missing[total_missing == total_rows].index, axis=1)

print(new_df.shape)

(21391, 79)
(21391, 77)


# Convert price column to float

### replacing '$', and ',' with ''

In [15]:
new_df['price'] = pd.to_numeric(new_df['price'].str.replace('$','', regex=False).str.replace(',','', regex=False), errors='coerce')

print(new_df['price'].dtype)
print(new_df['price'].describe())

float64
count    15398.000000
mean       176.100857
std        310.515222
min         11.000000
25%         75.000000
50%        128.000000
75%        207.000000
max      16829.000000
Name: price, dtype: float64


### Filter out listings with minimum nights required over 30 nights
### Filter out listings with no price data

In [16]:
new_df = new_df[new_df['minimum_nights'] <= 30]
new_df = new_df[~new_df['price'].isnull()]

print(new_df.shape)
print(new_df['minimum_nights'].max())
print(new_df['price'].isnull().sum())


(14427, 77)
30
0


# Export the cleaned dataset as a csv for later use

In [17]:
new_df.to_csv('listings_clean.csv', index=False)

## Cleaning Summary

**Columns dropped:** Removed `neighbourhood_group_cleansed` and `calendar_updated` 
as both were 100% empty and would provide no analytical value.

**Minimum nights filter:** Capped at 30 nights based on the mean (30) and median (30) 
of the column. Listings requiring longer stays are likely long-term rentals, not 
short-term Airbnb listings, and would skew the analysis.

**Missing price rows dropped:** 5,993 rows with no price data were removed. These 
listings are likely inactive or expired, and filling them with estimated values 
would introduce inaccurate data into a pricing analysis.

**Final dataset: 14,427 listings, 77 columns.**